os/arch/arm/src/amebasmart/amebasmart_idle.c

```diff
@@ -90,52 +90,42 @@ Theoretically, we will not expect PM state to drop to sleep again in such short
 But in case that happens, we have to introduce some strategy to handle that
 */
 RESLEEP_2:
-#endif
-		if (tizenrt_ready_to_sleep()) {
-#ifdef CONFIG_SMP
-			/*PG flow */
-			if (pmu_get_sleep_type() == SLEEP_PG) {
-				/* CPU1 is in task schedular, tell CPU1 to enter hotplug */
-				if (up_get_secondary_cpu_state(1) == CPU_RUNNING) {
-					/* CPU1 may in WFI idle state. Wake it up to enter hotplug itself */
-					up_irq_enable();
-					arm_cpu_sgi(GIC_IRQ_SGI4, (1 << 1));
-					arm_arch_timer_int_mask(0);
-					up_timer_enable();
-					DelayUs(100);
-					goto RESLEEP_1;
-				}
-				/* CPU1 just come back from pg, so can't sleep here */
-				if (up_get_secondary_cpu_state(1) == CPU_WAKE_FROM_SLEEP) {
-					pmdbg("Secondary core just woke from PG sleep!\n");
-					goto RESLEEP_2;
-				}
-				/* CG flow */
-			} else {
-				if (!check_wfi_state(1)) {
-					goto EXIT;
-				}
-			}
-#endif
-			/* Interrupt source will wake cpu up, just leave expected idle time as 0
-			Enter sleep mode for AP */
-			configPRE_SLEEP_PROCESSING(xModifiableIdleTime, handler);
-			/* When wake from pg, arm timer has been reset, so a new compare value is necessary to
-			trigger an timer interrupt */
-			if (pmu_get_sleep_type() == SLEEP_PG) {
+		/*PG flow */
+		if (pmu_get_sleep_type() == SLEEP_PG) {
+			/* CPU1 is in task schedular, tell CPU1 to enter hotplug */
+			if (up_get_secondary_cpu_state(1) == CPU_RUNNING) {
+				/* CPU1 may in WFI idle state. Wake it up to enter hotplug itself */
+				up_irq_enable();
+				arm_cpu_sgi(GIC_IRQ_SGI4, (1 << 1));
+				arm_arch_timer_int_mask(0);
 				up_timer_enable();
-				arm_arch_timer_set_compare(arm_arch_timer_count() + 50000);
+				DelayUs(100);
```

Comment:
why do we need DelayUs(100); here?

Reply:
@edwakuwaku ??

Reply:
Hello... This bsp idle logic will be revised in the PR: https://github.com/Samsung/TizenRT/pull/6256
The delay was added previously when the secondary core is failed to enter hotplug, as a kind of buffer time for secondary core to handle the hotplug signal....
As you know, the PM driver is keep on changing... Thus, BSP is forced to follow the PM driver change as well....
Please ignore this as of now as all of these logic will be revised later on...

Reply:
@gSahitya-samsung mark it with #todo to revise it later point of time.

Reply:
Done